osu.Game/Screens/Select/Carousel/DrawableCarouselBeatmapSet.cs

```diff
@@ -190,6 +180,48 @@ private void updateBeatmapDifficulties()
             }
         }
 
+        [Resolved]
+        private BeatmapCarousel.CarouselScrollContainer scrollContainer { get; set; } = null!;
+
+        private void loadContentIfRequired()
+        {
+            Quad containingSsdq = scrollContainer.ScreenSpaceDrawQuad;
+
+            // Using DelayedLoadWrappers would only allow us to load content when on screen, but we want to preload while off-screen
+            // to provide a better user experience.
+
+            // This is tracking time that this drawable is updating since the last pool.
+            // This is intended to provide a debounce so very fast scrolls (from one end to the other of the carousel)
+            // don't cause huge overheads.
+            //
+            // We increase the delay based on distance from centre, so the beatmaps the user is currently looking at load first.
+            float timeUpdatingBeforeLoad = 50 + Math.Abs(containingSsdq.Centre.Y - ScreenSpaceDrawQuad.Centre.Y) / containingSsdq.Height * 100;
+
+            Debug.Assert(Item != null);
+
+            if (loadCancellation == null && (timeSinceUnpool += Time.Elapsed) > timeUpdatingBeforeLoad)
+            {
+                loadCancellation = new CancellationTokenSource();
+
+                LoadComponentsAsync(new CompositeDrawable[]
+                {
+                    new SetPanelBackground(manager.GetWorkingBeatmap(beatmapSet.Beatmaps.MinBy(b => b.OnlineID)))
```

Comment:
The comment explaining why this is doing a `.MinBy()` is gone from the previous implementation. I would like for it to return because I was really confused why this was picking a random beatmap like that.

Reply:
Not sure how this vanished, have added back.